modules/imap/imap.go

```diff
@@ -17,15 +20,30 @@ type Connection struct {
 	Conn net.Conn
 }
 
+// Verify banner begins with a valid IMAP response and handle it
+func VerifyIMAPContents(n int, ret []byte) (string, error) {
+	s := string(ret[:n])
+	if strings.HasPrefix(s, "* OK"){
+		return s, nil
+	}
+	if strings.HasPrefix(s, "* NO"){
+		return s, zgrab2.NewScanError(zgrab2.SCAN_APPLICATION_ERROR, errors.New("IMAP reported error"))
```

Comment:
This isn't actually an error, at least for us.  Since the goal of zgrab2 is to identify services, receiving a well-formed error response is a good thing, as it is a positive indication that (in this case) we are talking to an imap server. (This also applies to the `* BAD` case, as well).

There is also `* PREAUTH` and `* BYE`